lib/data/repository/recipes_repository.dart

```diff
@@ -31,6 +32,40 @@ class RecipesRepository implements IRepository {
     }
   }
 
+  @override
+  Future<List<RecipeModel>> getSearchResults(
+    SearchDataObject parameters,
+  ) async {
+    List<RecipeModel> listOfRecipes = [];
+    try {
+      final response = await apiService.getSearchResult(
+        parameters.cuisine,
+        parameters.mealType,
+        parameters.maxCarbs,
+        parameters.maxProtein,
+        parameters.maxCalories,
+        parameters.maxFat,
+        parameters.maxCaffeine,
+        parameters.maxCopper,
+        parameters.maxCalcium,
```

Comment:
Instead to send all this parameters throw the method, as an improvement you can create a Map<String,String> as a body (Maybe could be call it "queryParameter") that is the name that you have inside apiService.
`queryParameters: {
        ApiConstants.numberQueryParameter: ApiConstants.numberOfSearchResults,
        'cuisine': cuisine,
        'mealType': mealType,
        'maxCarbs': maxCarbs,
        'maxProtein': maxProtein,
        'maxCalories': maxCalories,
        'maxFat': maxFat,
        'maxCaffeine': maxCaffeine,
        'maxCopper': maxCopper,
        'maxCalcium': maxCalcium,
      },`

Reply:
Done, a Map is created in the repository and passed to the apiService as a parameter.